handlers/load_test_preferences_handler.go

```diff
@@ -2,76 +2,207 @@
 package handlers
 
 import (
+	"encoding/json"
 	"net/http"
 	"strconv"
 	"time"
 
+	"github.com/golang/protobuf/jsonpb"
+
+	duration "github.com/golang/protobuf/ptypes/duration"
 	"github.com/layer5io/meshery/models"
+	SMPS "github.com/layer5io/service-mesh-performance-specification/spec"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+type custTestConf struct {
+	Val *SMPS.PerformanceTestConfig
+}
+
+func (c *custTestConf) MarshalJSON() ([]byte, error) {
+	m := jsonpb.Marshaler{
+		EmitDefaults: true,
+	}
+	val, err := m.MarshalToString(c.Val)
+	return []byte(val), err
+}
+
 // LoadTestPrefencesHandler is used for persisting load test preferences
 func (h *Handler) LoadTestPrefencesHandler(w http.ResponseWriter, req *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
-	if req.Method != http.MethodPost {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	q := req.FormValue("qps")
-	qps, err := strconv.Atoi(q)
-	if err != nil {
-		err = errors.Wrap(err, "unable to parse qps")
-		logrus.Error(err)
-		http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
-		return
-	}
-	if qps < 0 {
-		http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
-		return
-	}
-	dur := req.FormValue("t")
-	if _, err = time.ParseDuration(dur); err != nil {
-		err = errors.Wrap(err, "unable to parse t as a duration")
-		logrus.Error(err)
-		http.Error(w, "please provide a valid value for t", http.StatusBadRequest)
-		return
-	}
-	cu := req.FormValue("c")
-	c, err := strconv.Atoi(cu)
-	if err != nil {
-		err = errors.Wrap(err, "unable to parse c")
-		logrus.Error(err)
-		http.Error(w, "please provide a valid value for c", http.StatusBadRequest)
-		return
-	}
-	if c < 0 {
-		http.Error(w, "please provide a valid value for c", http.StatusBadRequest)
-		return
-	}
-	gen := req.FormValue("gen")
-	genTrack := false
-	// TODO: after we have interfaces for load generators in place, we need to make a generic check, for now using a hard coded one
-	for _, lg := range []models.LoadGenerator{models.FortioLG, models.Wrk2LG} {
-		if lg.Name() == gen {
-			genTrack = true
+
+	if req.Method == http.MethodPost || req.Method == http.MethodPut {
```

Comment:
Are we at risk of conflating what may become two separate actions for these different verbs in the future? What would it look like if we responded to one of them with a `501 Not Implemented`?

Reply:
Hmm, we could. Actually we do actually update the test config. Do you think we should remove support for that?